### PR TITLE
Feature/plnet dyad ranker

### PR DESCRIPTION
**Pull request by JonasHanselle**
This pull request contains the PLNetDyadRanker implementation and all associated tests and configurations.

### CODE REVIEW

**Review by mibraun — APPROVED** [review]
I added some small fixes and cosmetic improvements.
Otherwise I approve.

File: JAICore/jaicore-ml/test/jaicore/ml/dyadranking/general/AdvancedDyadDatasetDyadRankerTester.java [hunk 1]
```diff
@@ -41,30 +35,66 @@ public AdvancedDyadDatasetDyadRankerTester(ADyadRanker ranker) {
...
+				} 
+				currentMin = dyad;
+			}
+			avgFailures += failures;
 		}
-		System.out.println("Found failures: "+ failures);
+		avgKendallTau /= nTestInstances;
+		avgFailures /= nTestInstances;
+		
+		System.out.println("Kendall's tau: " + avgKendallTau);
```
Comment (helegraf) [inline JAICore/jaicore-ml/test/jaicore/ml/dyadranking/general/AdvancedDyadDatasetDyadRankerTester.java:83]: We have a class that is made to use multilabel measures that are already implemented for the context of dyad ranking: 
DyadRankingMLLossFunctionWrapper in jaicore.ml.dyadranking.loss can use all the ADecomposableDoubleMeasures from jaicore.ml.core.evaluation.measure.multilabel so if we add a measure here maybe it should be added as a multilabel measure? + the already implemented measures could possibly be used here for additional output.

File: JAICore/jaicore-ml/src/jaicore/ml/dyadranking/algorithm/PLNetDyadRanker.java [hunk 2]
```diff
@@ -1,43 +1,381 @@
...
+	 * zip file and a ".zip" ending is added.
+	 * 
+	 * @param filePath The file path to save to.
+	 * @throws IOException
+	 */
+	public void saveModelToFile(String filePath) throws IOException {
+		if (plNet == null) {
+			throw new IllegalStateException("Cannot save untrained model.");
+		}
+		File locationToSave = new File("filePath" + ".zip");
```
Comment (helegraf) [inline JAICore/jaicore-ml/src/jaicore/ml/dyadranking/algorithm/PLNetDyadRanker.java]: This does not actually zip the file,[ ModelSerializer.writeModel just uses a BufferedOutputStream](https://github.com/deeplearning4j/deeplearning4j/blob/master/deeplearning4j/deeplearning4j-nn/src/main/java/org/deeplearning4j/util/ModelSerializer.java ). To zip a file use a ZipOutPutStream ([Tutorial](https://www.baeldung.com/java-compress-and-uncompress)), or just change the file extension to not be zip.

**Review by helegraf — CHANGES_REQUESTED** [review]
Nice work; good docu + tests, from my comments mainly the .zip issue should be addressed the other one is more a suggestion.

File: JAICore/jaicore-ml/src/jaicore/ml/dyadranking/algorithm/PLNetDyadRanker.java (hunk 2 above)
Comment (helegraf) [reply JAICore/jaicore-ml/src/jaicore/ml/dyadranking/algorithm/PLNetDyadRanker.java]: Seems to work without actually using the zip stream, my bad

File: JAICore/jaicore-ml/test/jaicore/ml/dyadranking/general/AdvancedDyadDatasetDyadRankerTester.java (hunk 1 above)
Comment (mibraun) [reply JAICore/jaicore-ml/test/jaicore/ml/dyadranking/general/AdvancedDyadDatasetDyadRankerTester.java:83]: Should definitely be done in the future. However, the class does not exist on this branch yet.

**Review by helegraf — APPROVED** [review]

**Review by mirkojuergens — APPROVED** [review]

**Pull request merged**